lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -2527,34 +2528,77 @@ def _parse_comparisons(self):
     def __init__(self, **kwargs):
         super(AnsibleDockerClientContainer, self).__init__(**kwargs)
 
-        docker_api_version = self.version()['ApiVersion']
-        init_supported = LooseVersion(docker_api_version) >= LooseVersion('1.25')
-        if self.module.params.get("init") and not init_supported:
-            self.fail('docker API version is %s. Minimum version required is 1.25 to set init option.' % (docker_api_version,))
-
-        init_supported = init_supported and LooseVersion(docker_version) >= LooseVersion('2.2')
-        if self.module.params.get("init") and not init_supported:
-            self.fail("docker or docker-py version is %s. Minimum version required is 2.2 to set init option. "
-                      "If you use the 'docker-py' module, you have to switch to the docker 'Python' package." % (docker_version,))
-
-        uts_mode_supported = LooseVersion(docker_version) >= LooseVersion('3.5')
-        if self.module.params.get("uts") is not None and not uts_mode_supported:
-            self.fail("docker or docker-py version is %s. Minimum version required is 3.5 to set uts option. "
-                      "If you use the 'docker-py' module, you have to switch to the docker 'Python' package." % (docker_version,))
+        # Helper function to detect whether any specified network uses ipv4_address or ipv6_address
+        def detect_ipvX_address_usage():
+            for network in self.module.params.get("networks", []):
+                if 'ipv4_address' in network or 'ipv6_address' in network:
+                    return True
+            return False
 
-        blkio_weight_supported = LooseVersion(docker_version) >= LooseVersion('1.9')
-        if self.module.params.get("blkio_weight") is not None and not blkio_weight_supported:
-            self.fail("docker or docker-py version is %s. Minimum version required is 1.9 to set blkio_weight option.")
+        self.option_minimal_versions = dict(
+            init=dict(docker_py_version='2.2.0', docker_api_version='1.25'),
+            uts=dict(docker_py_version='3.5.0'),
+            blkio_weight=dict(docker_py_version='1.9.0'),
+            cpuset_mems=dict(docker_py_version='2.3.0'),
+            ipvX_address_supported=dict(docker_py_version='1.9.0', detect_usage=detect_ipvX_address_usage,
+                                        usage_msg='ipv4_address or ipv6_address in networks'),
+            runtime=dict(docker_py_version='2.4.0', docker_api_version='1.12'),
+            healthcheck=dict(docker_py_version='2.0.0'),
```

Comment:
@felixfontein I just check [docker api changes](https://docs.docker.com/engine/api/version-history/#v125-api-changes), healthcheck requires minimum docker api version is `1.25`. I missed condition for minimum docker api version. Could you please add it now?

Reply:
Done! I think we need to go through all existing options anyway to make sure we got the correct minimal API version and minimal docker-py version for all options. I wouldn't be surprised if some of the current checks are not completely correct, or haven't been there in the first place.